src/jenesis/contracts/schema.py

```diff
@@ -75,6 +75,19 @@ def generate_schemas(
 
     for contract in contracts:
         contract.schema = load_contract_schema(contract.source_path)
+        # check for workspace-style schemas
+        if contract.name in contract.schema:
+            contract_schema = contract.schema[contract.name]
+        else:
+            contract_schema = contract.schema
+
+        for (msg_type, schema) in contract_schema.items():
+            if 'instantiate' in msg_type:
+                contract.instantiate_schema = schema
+            elif 'query' in msg_type:
+                contract.query_schema = schema
+            elif 'execute' in msg_type:
+                contract.execute_schema = schema
```

Comment:
I think rather than duplicating this code, let's move this from here and from `Contract.__post_init()` to a new function in the Contract class, something like `Contract.update_schema(self)` which would then be called from both places.

Reply:
Sounds good